hubblestack/files/hubblestack_nova/mount.py

```diff
@@ -0,0 +1,223 @@
+                    ret['Controlled'].append(tag_data)
+                    continue
+                
+
+                name  = tag_data.get('name')
+                audittype = tag_data.get('type')
+
+                if 'attribute' not in tag_data:
+                    log.error('No attribute found for mount audit {0}, file {1}'
+                              .format(tag,name))
+                    tag_data = copy.deepcopy(tag_data)
+                    # don't know what this statement does. Adding it to keep the code similar to other modules
+                    tag_data['error'] = 'No pattern found'.format(mod) 
+                    ret['Failure'].append(tag_data)
+                    continue
+
+                attribute = tag_data.get('attribute')
+
+                # whether to check the existence of the path or not. If path_check is True and there is no partition on the specified path, 
+                # test case is marked as failure
+                check_type = 'hard'
+                if 'check_type' in tag_data:
+                    check_type = tag_data.get('check_type')
+
+                if check_type not in ['hard','soft']:
+                    log.error('Unrecognized option:' + check_type)
+                    tag_data = copy.deepcopy(tag_data)
+                    # don't know what this statement does. Adding it to keep the code similar to other modules
+                    tag_data['error'] = 'No valid check_type found'
+                    ret['Failure'].append(tag_data)
+                    continue
+
+                found = _check_mount_attribute(name,attribute,check_type)
+
+                if audittype == 'blacklist':
+                    if found:
+                        ret['Failure'].append(tag_data)
+                    else:
+                        ret['Success'].append(tag_data)
+
+                elif audittype == 'whitelist':
+                    if found:
+                        ret['Success'].append(tag_data)
+                    else:
+                        ret['Failure'].append(tag_data)
+
+    return ret
+
+
+def _merge_yaml(ret, data, profile=None):
+    '''
+    Merge two yaml dicts together at the grep:blacklist and grep:whitelist level
+    '''
+    if 'mount' not in ret:
+        ret['mount'] = {}
+    for topkey in ('blacklist', 'whitelist'):
+        if topkey in data.get('mount', {}):
+            if topkey not in ret['mount']:
+                ret['mount'][topkey] = []
+            for key, val in data['mount'][topkey].iteritems():
+                if profile and isinstance(val, dict):
+                    val['nova_profile'] = profile
+                ret['mount'][topkey].append({key: val})
+    return ret
+
+
+def _get_tags(data):
+    '''
+    Retrieve all the tags for this distro from the yaml
+    '''
+
+    ret = {}
+    distro = __grains__.get('osfinger')
+
+    for toplist, toplevel in data.get('mount', {}).iteritems():
+        # grep:blacklist
```

Comment:
It should be mount and not grep. and we have such cases at many places. Please correct all of these.